src/webpack/index.ts

```diff
@@ -1,103 +1,110 @@
-import url from 'url';
 import { Compiler, compilation as webpackCompilation } from 'webpack';
+import { Asset, Manifest } from './manifest';
+import { buildManifest } from './utils';
 
-type Compilation = webpackCompilation.Compilation;
+type Parser = webpackCompilation.normalModuleFactory.Parser;
 
-type ModuleExtended = webpackCompilation.Module & {
-  libIdent?: (params: { context: any }) => string;
-  rootModule?: webpackCompilation.Module;
-  rawRequest?: string;
-};
-
-export type Manifest = { [key: string]: Bundle };
-
-export type Bundle = {
-  id: number | string | null;
-  file: string;
-  name: string | null;
-  publicPath: string;
-};
+export type { Asset, Manifest };
 
 export type ReactLooselyLazyPluginOptions = {
   filename: string;
 };
 
-function buildManifest(compiler: Compiler, compilation: Compilation) {
-  const { context } = compiler.options;
-  const manifest: Manifest = {};
-
-  compilation.chunks.forEach((chunk: webpackCompilation.Chunk) => {
-    chunk.files.forEach(file => {
-      chunk.getModules().forEach((module: ModuleExtended) => {
-        const id = module.id;
-        const name =
-          typeof module.libIdent === 'function'
-            ? module.libIdent({ context })
-            : null;
-        const publicPath = url.resolve(
-          compilation.outputOptions.publicPath || '',
-          file
-        );
-        let currentModule = module;
-
-        if (!name) {
+export class ReactLooselyLazyPlugin {
+  name: string = ReactLooselyLazyPlugin.name;
+  filename: string;
+
+  constructor(opts: ReactLooselyLazyPluginOptions) {
+    this.filename = opts.filename;
+  }
+
+  apply(compiler: Compiler) {
+    const { name } = this;
+    // A mapping of module paths to the set of raw requests that stem from the react-loosely-lazy library
+    const moduleImports = new Map<string, Set<string>>();
+
+    const onJavaScriptModule = (parser: Parser) => {
+      let currentLibraryExpression: undefined | any;
+      let hasLibraryImport = false;
+
+      parser.hooks.import.tap(name, (statement, source) => {
+        if (source === 'react-loosely-lazy') {
```

Comment:
Nit: would be nice to extract this string to some const somewhere as needed in several places (webpack, babel, maybe ESLint in future...)

Reply:
Yep, this can be done in a future PR :)